tcms/kiwi_auth/tests/test_views.py

```diff
@@ -262,3 +283,26 @@ def test_confirm(self):
         self.assertTrue(user.is_active)
         activate_key_deleted = not UserActivationKey.objects.filter(user=user).exists()
         self.assertTrue(activate_key_deleted)
+
+
+class TestLoginViewWithCustomTemplate(TestCase):
+    """Test for login view with custom template"""
+
+    def test_get_template_names(self):
+        try:
+            response = self.client.get(reverse("tcms-login"))
+            self.assertIsNotNone(response.template_name)
+            self.assertEqual(
+                response.template_name,
+                ["registration/custom_login.html", "registration/login.html"],
+            )
+        except TemplateDoesNotExist:
+            self.fail("Failed to load template")
```

Comment:
Why do we need a try-except here ? From what I can tell if there's an uncaught exception the test will report FAIL/ERROR. OTOH if the templates don't match then the assertion will fail. 

I can't figure out in which case you want to catch a TemplateDoesNotExist exception and then report a failed test and that would be a good thing.

Reply:
I thought of considering the case when templates don't match, in that case TemplateDoesNotExist is raised. But now I realize that if the templates don't match, then the assertion will fail, so adding try-except here don't make sense. I have removed it now.